apps/spruce/src/pages/image/useEvents.ts

```diff
@@ -0,0 +1,23 @@
+import { useState } from "react";
+
+export const IMAGE_EVENT_LIMIT = 5;
+
+export const useEvents = () => {
+  const [allEventsFetched, setAllEventsFetched] = useState(false);
+  const [prevCount, setPrevCount] = useState(0);
+
+  // Hide Load More button when event count < limit is returned,
+  // or when an additional fetch fails to load more events.
+  const onCompleted = (count: number) => {
+    if (count - prevCount < IMAGE_EVENT_LIMIT) {
+      setAllEventsFetched(true);
+    }
+  };
+
+  return {
+    allEventsFetched,
+    setAllEventsFetched,
+    onCompleted,
+    setPrevCount,
+  };
+};
```

Comment:
Not exactly sure how feasible this would be because the distro/project settings eventLog page useEvents hook uses the limit field as an argument, while I use the IMAGE_EVENT_LIMIT as the limit which is defined on the page, so the hooks are slightly different.

Reply:
Can you pass the limit to the hook though? it seems like that would be possible.

Reply:
Got it - that makes a lot of sense!